.github/workflows/dependencies-head.yml

```diff
@@ -140,12 +144,13 @@ jobs:
 
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip setuptools wheel
-        python -m pip --no-cache-dir --quiet install --upgrade ".[all,test]"
-        python -m pip uninstall --yes matplotlib
+        python -m pip install uv
+        uv pip install --system --upgrade pip setuptools wheel
+        uv pip --no-cache --quiet install --system --upgrade ".[all,test]"
+        uv pip uninstall --yes matplotlib
         # Need to use --extra-index-url as dependencies aren't on scientific-python-nightly-wheels package index.
         # Need to use --pre as dev releases will need priority over stable releases.
-        python -m pip install \
+        uv pip install --system \
```

Comment:
Double check this as `pip` and `uv` handle `--extra-index-url` differently. c.f. https://github.com/scientific-python/upload-nightly-action/issues/76

Reply:
Yeah, this needed to get flipped to

```toml
        uv pip install --system \
          --pre \
          --index-url https://pypi.org/simple/ \
          --extra-index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple \
          matplotlib
```

as `uv` give priority _to `--extra-index-url`_.

```
$ pip list
Package         Version
--------------- -------------------------
contourpy       1.3.0.dev1
cycler          0.12.1
fonttools       4.51.0
kiwisolver      1.4.5
matplotlib      3.10.0.dev114+g4cbef2d4c0
numpy           2.1.0.dev0
packaging       24.0
pillow          10.3.0
pip             24.0
pyparsing       3.1.2
python-dateutil 2.9.0.post0
setuptools      69.5.1
six             1.16.0
uv              0.1.39
wheel           0.43.0
```